src/ring_theory/integral_closure.lean

```diff
@@ -357,6 +357,63 @@ begin
   exact hp',
 end
 
+/-- If the integral extension `R → S` is injective, and `S` is a field, then `R` is also a field -/
```

Comment:
@Vierkantor To make sure I'm understanding this right, the idea would be to take `p' = -div_X (reverse p)`, and then prove that `(algebra_map R S a)⁻¹ = aeval (algebra_map R S a) p' = algebra_map R S (eval a p')`, and then just apply injectivity to get that `eval a p'` is an inverse? 
I think the main dificulty would be proving that `(algebra_map R S a)⁻¹ - aeval (algebra_map R S a) p' = (algebra_map a)^(n-1) * aeval a^-1 p =  0', but `rec_on_horner` would hopefully make this easier.

@jcommelin I think in the thread @Vierkantor mentioned  `at_infinity` was discussed as a version of `reverse`, but it doesn't look like it was added yet. It seems like you were somewhat involved in the discussion, do you know if this might get added any time soon?

Reply:
I was thinking something like that, indeed. Note that I only looked at your proof and didn't try writing the new version, so it might be more difficult than I'm expecting.